storagedriver/azure/blockid.go

```diff
@@ -0,0 +1,56 @@
+package azure
+
+import (
+	"encoding/base64"
+	"fmt"
+	"math/rand"
+	"sync"
+
+	azure "github.com/MSOpenTech/azure-sdk-for-go/clients/storage"
+)
+
+type blockIDGenerator struct {
+	pool map[string]bool
+	m    sync.Mutex
+}
+
+// Generate returns an unused random block id and adds the generated ID
```

Comment:
If the pool is re-populated each time, why not use a deterministic ID allocation approach?